cranelift/isle/docs/language-reference.md

```diff
@@ -1458,45 +1460,43 @@ The grammar accepted by the parser is as follows:
 <ISLE> ::= <def>*
 
 <def> ::= "(" "pragma" <pragma> ")"
-        | "(" "type" <typedecl> ")"
+        | "(" "type" <type> ")"
         | "(" "decl" <decl> ")"
         | "(" "rule" <rule> ")"
-        | "(" "extractor" <etor> ")"
+        | "(" "extractor" <extractor> ")"
         | "(" "extern" <extern> ")"
         | "(" "convert" <converter> ")"
 
-// No pragmas are defined yet
+;; No pragmas are defined yet
 <pragma> ::= <ident>
 
-<typedecl> ::= <ident> [ "extern" | "nodebug" ] <typevalue>
```

Comment:
technically not a *value*, could be confusing

Reply:
Well, here I would disagree again. You can call it a "type expression" if you want, but it's a value in the universe of types. It is definitely not a `typedef`: `typedef` is, in my mind, another name for the whole form that defines a new type (like a typedef in C/C++ for example). Perhaps "type body" if you want to distinguish the semantic place this occupies -- an expanded form that can be named in a `type` declaration form's right-hand side but not in-line wherever types are given.

Reply:
Yeah, typedef is worse than typevalue.. I did mean something like "type body".